var/spack/repos/builtin/packages/py-dask-mpi/package.py

```diff
@@ -17,6 +17,7 @@ class PyDaskMpi(PythonPackage):
     version('2.21.0', sha256='76e153fc8c58047d898970b33ede0ab1990bd4e69cc130c6627a96f11b12a1a7')
     version('2.0.0', sha256='774cd2d69e5f7154e1fa133c22498062edd31507ffa2ea19f4ab4d8975c27bc3')
 
-    depends_on('py-dask@2.2:', type=('build', 'run'))
     depends_on('py-mpi4py', type=('build', 'run'))
     depends_on('py-setuptools', type='build')
+    for ver in ('@2.0.0', '@2.21.0'):
+        depends_on('py-dask{}'.format(ver), type=('build', 'run'), when=ver)
```

Comment:
It seems that `dask-mpi` doesn't follow the same versions of `dask`, and it does not have these restrictions in their requirements:
- https://github.com/dask/dask-mpi/blob/2.0.0/environment.yml
- https://github.com/dask/dask-mpi/blob/2.21.0/environment.yml
- https://github.com/dask/dask-mpi/blob/main/environment.yml

Wouldn't work to add an explicit requirement for `py-dask` in `var/spack/repos/builtin/packages/py-sonata-network-reduction/package.py`, that seems consistent with the `setup.py` in `sonata-network-reduction`?
```
    depends_on('py-dask@2.0:2.21', type='run', when='@0.1.2:')
```

Reply:
nice catch, thank you